src/mapeo-project.js

```diff
@@ -548,6 +565,59 @@ export class MapeoProject extends TypedEmitter {
   get $icons() {
     return this.#iconApi
   }
+
+  async $leave() {
+    await assertDeviceCanLeaveProject(this.#deviceId, this.#capabilities)
+
+    // update client database (delete all encryption keys except auth)
+    const encoded = ProjectKeys.encode({
+      projectKey: Buffer.from(this.#projectId, 'hex'),
+      projectSecretKey: this.#projectSecretKey,
+      encryptionKeys: { auth: this.#encryptionKeys.auth },
+    }).finish()
+
+    const nonce = projectIdToNonce(this.#projectId)
+
+    // 1. Update relevant shared database tables
+    this.#sharedDb
+      .update(projectKeysTable)
+      .set({
+        keysCipher: this.#keyManager.encryptLocalMessage(
+          Buffer.from(encoded.buffer, encoded.byteOffset, encoded.byteLength),
+          nonce
+        ),
+      })
+      .where(eq(projectKeysTable.projectId, this.#projectId))
+      .run()
+
+    this.#sharedDb
+      .delete(projectSettingsTable)
+      .where(eq(projectSettingsTable.docId, this.#projectId))
+      .run()
```

Comment:
I _think_ this makes sense but not entirely sure. My assumption is that once you leave a project, you shouldn't be able to access this kind of information, but maybe that's too strict in this case?